compiler/hash-typecheck/src/ops/validate.rs

```diff
@@ -14,11 +22,56 @@ use crate::{
 
 /// Represents the level of a term.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+/// A enumeration of the level kinds that terms can be.
 pub enum TermLevel {
+    /// Couldn't be determined and thus labelled as unknown
+    Unknown,
+    /// Level 0 terms
     Level0,
+    /// Level 1 terms
     Level1,
+    /// Level 2 terms
     Level2,
+    /// Level 3 terms
     Level3,
+    /// Higher order levels
+    LevelN,
+}
+
+impl Display for TermLevel {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TermLevel::Unknown => write!(f, "unknown"),
+            TermLevel::Level0 => write!(f, "level-0"),
+            TermLevel::Level1 => write!(f, "level-1"),
+            TermLevel::Level2 => write!(f, "level-2"),
+            TermLevel::Level3 => write!(f, "level-3"),
+            TermLevel::LevelN => write!(f, "level-n"),
+        }
+    }
+}
+
+impl Term {
+    /// Compute the level of the term. This is a primitive computation
+    /// and does not attempt to compute the true level of the [Term]
+    /// by looking at the inner children of the [Term].
+    pub fn get_term_level(&self) -> TermLevel {
```

Comment:
No reason, I think it can be made to recurse in the same way as `NameFieldOrigin::new_from_term`? Is it possible to always get a known level from a term